src/App.tsx

```diff
@@ -1,17 +1,59 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import { getTodos } from './api/todos';
+
+import { SortType } from './types/SortType';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortType, setSortType] = useState<SortType>(SortType.All);
+  const [error, setError] = useState(false);
+
+  const filteredTodo = [...todos].filter(todo => {
+    switch (sortType) {
+      case SortType.Active:
+        return !todo.completed;
+      case SortType.Completed:
+        return todo.completed;
+      default:
+        return todo;
+    }
+  });
```

Comment:
I think it's better to save todos to the state, and just update the state after filtering

Reply:
this filtred array goes as a props to another component :)

Reply:
will be better to cover it with useMemo